src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -215,6 +215,13 @@ private static OnnxTransformer Create(IHostEnvironment env, ModelLoadContext ctx
                 RecursionLimit = recursionLimit
             };
 
+            IHostEnvironmentInternal localEnvironment = env as IHostEnvironmentInternal;
```

Comment:
Do we want this flag to work even if they aren't loading the model? Like if they set these flags should it work when they create the model initially as well?

Reply:
I made all ApplyOnnxModel overloads to use `MLContext.GpuDeviceId` when having the default `gpuDeviceId` parameter.